modules/arm_plugin/src/arm_executable_network.cpp

```diff
@@ -52,6 +53,15 @@ ArmPlugin::ExecutableNetwork::CreateInferRequestImpl(InferenceEngine::InputsData
                                              networkOutputs,
                                              std::static_pointer_cast<ExecutableNetwork>(shared_from_this()));
 }
+InferenceEngine::IInferRequestInternal::Ptr
+ArmPlugin::ExecutableNetwork::CreateInferRequestImpl(const std::vector<std::shared_ptr<const ov::Node>>& inputs,
+                                                     const std::vector<std::shared_ptr<const ov::Node>>& outputs) {
+    if (!this->_plugin || !this->_plugin->GetCore() || !this->_plugin->GetCore()->isNewAPI())
```

Comment:
Is it expected behaviour or maybe we should throw "NotImplemented" here?

Reply:
I don't think it is right to throw "Not implemented" because the method is actually implemented, but use of it is unexpected by the core.
Anyway call to the method is guarded with "try ... catch" so the exception will be silently disposed and only the null value of the pointer is actually checked

Reply:
Return of `nullptr` will work too for code from openvino/inference-engine/src/inference_engine/src/cpp_interfaces/interface/ie_iexecutable_network_internal.cpp:60.
But it is expected that the plugin will throw `NotImplemented` if something is not implemented.
And it is intended that the catch of this exception will leave `asyncRequestImpl`  not initialized.

Reply:
My point is that it's confusing to return not implemented in case functionality is actually implemented.
The fault happens somewhere deep in openvino sources, but the root cause is successful execution of method that is expected to be unsuccessful and shouldn't be even called. May be it would be better to throw `Unexpected`?

Reply:
@ilya-lavrenov @ilyachur  We need help!

Reply:
let's just implement this method as for all other plugins.

Reply:
Method is implemented. But its execution cause failure of subsequent stages if IE core is configured for old API. The question is how method should behave in case it's called despite the fact old API is configured?

Reply:
Looks like we can keep as is
The code which handles this logic can work with both nullptr, exception.

See https://github.com/openvinotoolkit/openvino/commit/0f3c10c810d953b82c81c6f4f1faf5f56aa90195#diff-2d02d4c6502f0eba954faa430284fffe4acd8de85722907085de9f9464b01688R61-R67